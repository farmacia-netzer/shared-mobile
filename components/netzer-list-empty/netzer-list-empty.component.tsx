import { BASE_MARGIN, BASE_PADDING } from '##theme/dimensions.constant';
import { GRAY_SCALE } from '##theme/grayscale.constant';
import { FONT_SIZE } from '##theme/typography.constant';
import React, { FC } from 'react';
import { StyleSheet, View } from 'react-native';
import { NetzerText } from '../netzer-text';

interface NetzerListEmptyProps {
  Image: Function | null;
  text: string;
}

export const NetzerListEmpty: FC<NetzerListEmptyProps> = ({ Image, text }) => {
  return (
    <View style={styles.empty}>
      {Image && <Image />}
      <NetzerText style={styles.emptyText} text={text} fontStyle={'BOLD'} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flex: 1,
    marginTop: BASE_MARGIN,
    paddingHorizontal: BASE_PADDING
  },
  empty: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  emptyText: {
    fontSize: FONT_SIZE.LARGE,
    marginTop: 30,
    color: GRAY_SCALE.GRAY_50
  }
});
