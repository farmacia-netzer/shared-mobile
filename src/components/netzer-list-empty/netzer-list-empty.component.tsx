
import { GRAY_SCALE } from '../../theme/grayscale.constant';
import { FONT_SIZE } from '../../theme/typography.constant';
import React, { FC, useContext, useMemo } from 'react';
import { StyleSheet, View } from 'react-native';
import { NetzerText } from '../netzer-text';
import { ThemeContext, ThemeI } from '../../context/theme/theme-context';

interface NetzerListEmptyProps {
  Image: Function | null;
  text: string;
  description?: string;
}

export const NetzerListEmpty: FC<NetzerListEmptyProps> = ({ Image, text, description }) => {
  const { theme } = useContext(ThemeContext);
  const styles = useMemo(() => stylesComponent(theme), [theme])

  return (
    <View style={styles.empty}>
      {Image && <Image />}
      <NetzerText style={styles.emptyText} text={text} fontStyle={'BOLD'} type="TITLE" />
      {description && <NetzerText style={styles.description} text={description} fontStyle={'BOLD'} />}
    </View>
  );
};

const stylesComponent = (theme: ThemeI) => StyleSheet.create({
  empty: {
    flex: 1,
    alignItems: 'center',
    paddingHorizontal: 30,
    justifyContent: 'center'
  },
  emptyText: {
    fontSize: FONT_SIZE.LARGE,
    marginTop: 30,
    textAlign: "center",
    color: theme.colors.text
  },
  description: {
    marginTop: 10,
    fontWeight: "500",
    color: GRAY_SCALE.GRAY_70,
    textAlign: "center",
    lineHeight: 25
  }
});
