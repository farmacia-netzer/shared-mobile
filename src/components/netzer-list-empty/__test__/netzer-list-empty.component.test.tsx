import { render } from '@testing-library/react-native';
import React from 'react';

import { NetzerListEmpty } from '../netzer-list-empty.component';

describe('<NetzerListEmpty />', () => {
  it('renders', () => {
    const { toJSON } = render(<NetzerListEmpty Image={null} text="hello" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
