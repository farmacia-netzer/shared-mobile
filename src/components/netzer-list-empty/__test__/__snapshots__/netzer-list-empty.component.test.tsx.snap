// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NetzerListEmpty /> renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    fontStyle="BOLD"
    style={
      Array [
        Object {
          "fontSize": 14,
          "lineHeight": 18,
        },
        undefined,
        Object {
          "color": "#767676",
          "fontSize": 18,
          "marginTop": 30,
        },
      ]
    }
  >
    hello
  </Text>
</View>
`;
